### PR TITLE
Reorganize validation test packages

### PR DESCRIPTION
**Pull request by marchof**
* Package names correspond module names
* Filter validation tests and targets in same package
* Move non-version specific tests to org.jacoco.core.test

### CODE REVIEW

**Comment by marchof** [comment]
@Godin Can you have a look please?

**Comment by marchof** [comment]
@Godin Just realized this... I tested locally with JDK 8 only 😢

**Comment by marchof** [comment]
@Godin I reverted `FramesTest`to its previous scope.

File: org.jacoco.core.test.validation.java5/src/org/jacoco/core/test/validation/java5/ConstructorsTest.java [hunk 1]
```diff
@@ -9,21 +9,21 @@
...
-import org.jacoco.core.test.filter.targets.Constructor;
 import org.jacoco.core.test.validation.ValidationTestBase;
+import org.jacoco.core.test.validation.java5.targets.ConstructorsTarget;
 import org.junit.Test;
 
 /**
- * Test of filtering of a bytecode that is generated for a private empty
- * constructors that do not have no arguments.
+ * Tests for different constructors. Private empty constructors that do not have
+ * no arguments are filtered.
```
[the remark this answers is not in the document]
Comment (marchof) [reply org.jacoco.core.test.validation.java5/src/org/jacoco/core/test/validation/java5/ConstructorsTest.java]: Fixed.

**Comment by marchof** [comment]
@Godin I fixed the commit comment and Javadoc.

When I looked at our test targets I realized that almost all of our "old" test targets were affected by filters. Some of the old test targets (like `Target07`) were completely covered by the new filter tests (`ConstructorTargets`). With more and more filters it will become more difficult to create a separate test for each filter.

For analysis and development I have a different setup in mind: I would like to introduce a (optional) wrapper for filters which keeps track of the filter activity. With this we can create a report for our validation tests which filter was active on what source file and line.

**Comment by Godin** [comment]
> wrapper for filters which keeps track of the filter activity

Yep, sounds good and we already discussed/mentioned something like this in past.

**Pull request merged**